### PR TITLE
Workspace updates

### PR DESCRIPTION
**Pull request by timmillwood**
Not ready for merge yet

### CODE REVIEW

File: src/Entity/Workspace.php [hunk 1]
```diff
@@ -50,7 +51,19 @@ class Workspace extends ContentEntityBase implements WorkspaceInterface {
...
+      ->setReadOnly(TRUE)
+      ->setSetting('unsigned', TRUE);
+
+    $fields['label'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Workaspace ID'))
+      ->setDescription(t('The workspace ID.'))
+      ->setSetting('max_length', 128)
+      ->setRequired(TRUE);
+
+    $fields['machine_name'] = BaseFieldDefinition::create('string')
```
Comment (dickolsson) [inline src/Entity/Workspace.php]: I would rename this to just `name`, it's shorter and simpler. The `machine_` part doesn't make any sense here :)

Comment (timmillwood) [inline src/Entity/Workspace.php]: We decided last week that using machine_name was more explicit and therefore the better option.

File: src/Entity/Workspace.php [hunk 2]
```diff
@@ -50,9 +51,21 @@ class Workspace extends ContentEntityBase implements WorkspaceInterface {
...
+      ->setReadOnly(TRUE)
+      ->setSetting('unsigned', TRUE);
+
+    $fields['label'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Workaspace ID'))
+      ->setDescription(t('The workspace label.'))
+      ->setSetting('max_length', 128)
+      ->setRequired(TRUE);
+
+    $fields['machine_name'] = BaseFieldDefinition::create('string')
```
Comment (jeqq) [inline src/Entity/Workspace.php]: Why do we need this field if we have `id` (an unique identifier for the workspace) and `label`? I see the necessity of a machine_name field when it is used as an unique identifier but in this case we already have the `id` field.

Comment (timmillwood) [inline src/Entity/Workspace.php]: id - a numeric identifier for the workspace
machine_name - a machine readable name and url parameter for the workspace
label - a human readable name for the workspace

Comment (timmillwood) [inline src/Entity/Workspace.php]: Also, to add to that, the reasoning behind 3 separate fields is so the workspace can be renamed.

**Pull request merged**